client/go/outline/electron/vpn.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"log/slog"
+	"sync"
+
+	perrs "github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+)
+
+type vpnConfigJSON struct {
+	ID              string   `json:"id"`
+	InterfaceName   string   `json:"interfaceName"`
+	IPAddress       string   `json:"ipAddress"`
+	DNSServers      []string `json:"dnsServers"`
+	RoutingTableId  int      `json:"routingTableId"`
+	RoutingPriority int      `json:"routingPriority"`
+	ProtectionMark  uint32   `json:"protectionMark"`
+	TransportConfig string   `json:"transport"`
+}
+
+type vpnConnectionJSON struct {
+	ID       string `json:"id"`
+	Status   string `json:"status"`
+	RouteUDP *bool  `json:"routeUDP"`
+}
+
+type VPNStatus string
+
+const (
+	VPNConnected     VPNStatus = "Connected"
+	VPNDisconnected  VPNStatus = "Disconnected"
+	VPNConnecting    VPNStatus = "Connecting"
+	VPNDisconnecting VPNStatus = "Disconnecting"
+)
+
+type VPNConnection interface {
+	ID() string
+	Status() VPNStatus
+	RouteUDP() *bool
```

Comment:
Might be worth documenting this interface, most useful for `RouteUDP()` as it confused me: "route" implies action, but it's just a getter. How about `SupportsUDP()` or `CanRouteUDP()`?

Reply:
Updated.